src/project.js

```diff
@@ -0,0 +1,177 @@
+import { v4 as uuidv4 } from 'uuid';
+import Task from './task';
+
+class Project {
+  constructor(name) {
+    this.name = name;
+    this.tasks = [];
+    this.head_id = uuidv4();
+    this.body_id = uuidv4();
+  }
+}
+
+const setDisplayNone = (name, date, priority, button) => {
+  name.style.display = 'none';
+  date.style.display = 'none';
+  priority.style.display = 'none';
+  button.style.display = 'none';
+};
+
+const storeProjectInLocalStorage = (project) => {
+  let projects;
+  if (localStorage.getItem('projects') === null) {
+    projects = [];
+  } else {
+    projects = JSON.parse(localStorage.getItem('projects'));
+  }
+  projects.push(project);
+  localStorage.setItem('projects', JSON.stringify(projects));
+};
+
+const storeTaskInLocalStorage = (task, project) => {
+  const projects = JSON.parse(localStorage.getItem('projects'));
+
+  projects.forEach((proj) => {
+    if (project.head_id === proj.head_id) {
+      proj.tasks.push(task);
+
+      localStorage.removeItem('projects');
+      localStorage.setItem('projects', JSON.stringify(projects));
+    }
+  });
+};
+
+const removeProjectFromLocalStorage = (project) => {
+  const projects = JSON.parse(localStorage.getItem('projects'));
+
+  projects.forEach((proj) => {
+    if (project.head_id === proj.head_id) {
+      const index = projects.indexOf(proj);
+      projects.splice(index, 1);
+      localStorage.removeItem('projects');
+      localStorage.setItem('projects', JSON.stringify(projects));
+    }
+  });
+};
+
+const addNewProject = (() => {
+  const content = document.querySelector('#content');
+  const nameInput = document.createElement('input');
+  const addButton = document.createElement('button');
+  const div = document.createElement('div');
+  div.classList.add('d-flex');
+  div.appendChild(addButton);
+  div.appendChild(nameInput);
+
+  nameInput.classList.add('form-control');
+  nameInput.placeholder = 'Enter project name';
+  addButton.classList.add('btn', 'btn-primary', 'my-button');
+  addButton.innerText = '+';
+  content.appendChild(div);
+
+  // event listener for add project Button
+  addButton.addEventListener('click', () => {
+    if (nameInput.value === '') {
+      alert('Put the project name please');
```

Comment:
- [ ] Avoid using window.alert as it blocks the script running. Please remove this and other occurrences of alert and create an error div instead.